src/safeposix/vmmap.rs

```diff
@@ -0,0 +1,587 @@
+use super::vmmap_constants::*;
+use std::io;
+use nodit::NoditMap;
+use nodit::{interval::ie, Interval};
+
+/// Used to identify whether the vmmap entry is backed anonymously,
+/// by an fd, or by a shared memory segment
+
+#[allow(dead_code)]
+#[derive(Clone, Copy, PartialEq, Eq, Debug)]
+pub enum MemoryBackingType {
+    None, // just a dummy value for places where it needs to be passed, but you dont have the value
+    Anonymous,
+    SharedMemory(u64),   // stores shmid
+    FileDescriptor(u64), // stores file descriptor addr
+}
+
+/// In the old native client based vmmap, we relied on the fd, shmid
+/// fields. 
+/// 
+/// An entry in the virtual memory map that contains fields such as page number, number of pages, 
+/// permissions, file offset, file size, shared memory ID, and backing fields to distinguish memory types.
+#[derive(Clone, PartialEq, Eq, Debug)]
+pub struct VmmapEntry {
+    pub page_num: u32, /* base virtual addr >> NACL_PAGESHIFT */
+    pub npages: u32,   /* number of pages */
+    pub prot: i32,     /* mprotect attribute */
+    pub maxprot: i32,
+    pub flags: i32,       /* mapping flags */
+    pub removed: bool,    /* flag set in fn Update(); */
+    pub file_offset: i64, /* offset into desc */
+    pub file_size: i64,   /* backing store size */
+    pub cage_id: u64,
+    pub backing: MemoryBackingType,
+}
+
+#[allow(dead_code)]
+impl VmmapEntry {
+    pub fn new(
+        page_num: u32,
+        npages: u32,
+        prot: i32,
+        maxprot: i32,
+        flags: i32,
+        removed: bool,
+        file_offset: i64,
+        file_size: i64,
+        cage_id: u64,
+        backing: MemoryBackingType,
+    ) -> Self {
+        return VmmapEntry {
+            page_num,
+            npages,
+            prot,
+            maxprot,
+            flags,
+            removed,
+            file_offset,
+            file_size,
+            cage_id,
+            backing,
+        };
+    }
+
+    fn max_prot(&self) -> i32 {
```

Comment:
I believe this is the function to find the max protection from a file backed memory region, and is currently incomplete? @pranav-bhatt @ruchjoshi-nyu can you confirm?

Reply:
Yes, I believe it is incomplete

Reply:
change this function name to `get_max_prot()`

Reply:
@ruchjoshi-nyu change args to `(self, cageid: u64, fd, u64)`

Reply:
@Yaxuan-w can you change the contents of this function to call into fdtables with the cageid and fd, and return the file protection of that fd.